core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -149,7 +159,13 @@ class HistoryServer(
    *
    * @return List of all known applications.
    */
-  def getApplicationList(): Iterable[ApplicationHistoryInfo] = provider.getListing()
+  def getApplicationList(): Iterable[ApplicationHistoryInfo] = {
+    provider.getListing()
+  }
+
+  def getApplicationInfoList: Seq[ApplicationInfo] = {
+    getApplicationList().map { ApplicationsListResource.appHistoryInfoToPublicAppInfo }.toSeq
```

Comment:
Is this transformation lazy? Otherwise it can cause memory usage to shoot up considerably when this method is called (which can happen from multiple concurrent user requests).